interp/constrextern.ml

```diff
@@ -307,8 +307,20 @@ let extern_evar n l = CEvar (n,l)
     For instance, in the debugger the tables of global references
     may be inaccurate *)
 
+let rec dirpath_of_modpath = function
+  | MPfile dp -> dp
+  | MPbound mbid -> let (_,id,_) = MBId.repr mbid in DirPath.make [id]
+  | MPdot (t, l) -> Libnames.add_dirpath_suffix (dirpath_of_modpath t) (Label.to_id l)
+
+let path_of_global = function
+  | GlobRef.VarRef id -> Libnames.make_path DirPath.empty id
+  | GlobRef.ConstRef cst -> Libnames.make_path (dirpath_of_modpath (Constant.modpath cst)) (Label.to_id (Constant.label cst))
+  | GlobRef.IndRef (ind,_) -> Libnames.make_path (dirpath_of_modpath (MutInd.modpath ind)) (Id.of_string ("<unknown-inductive>" ^ Label.to_string (MutInd.label ind) ^ ">"))
+  | GlobRef.ConstructRef ((ind,_),_) -> Libnames.make_path (dirpath_of_modpath (MutInd.modpath ind)) (Id.of_string ("<unknown-constructor-of-" ^ Label.to_string (MutInd.label ind) ^ ">"))
+
 let default_extern_reference ?loc vars r =
-  Nametab.shortest_qualid_of_global ?loc vars r
+  try Nametab.shortest_qualid_of_global ?loc vars r
+  with Not_found when GlobRef.is_bound r -> qualid_of_path (path_of_global r)
```

Comment:
You should probably tweak `path_of_global` to return a qualid directly instead of going through this indirection.

Also, it might be useful to not check for boundedness, sometimes in the debugger we face the same issue even if the reference is absolute.

Reply:
> You should probably tweak `path_of_global` to return a qualid directly instead of going through this indirection.

You mean adding some `qualid_of_global` in the API? I could.

> Also, it might be useful to not check for boundedness, sometimes in the debugger we face the same issue even if the reference is absolute.

Currently, the debugger has its own printer. So, this is not necessary in this PR.

Reply:
> You mean adding some qualid_of_global in the API? I could.

I meant building the qualid directly with `Libnames.make_qualid` from a dirpath obtained through the nametab and the dummy identifier. But it's purely cosmetic, it does not matter anyways.

Reply:
The problem is that the dirpath is here not in the nametab.